docs/design/downward_api_resources_limits_requests.md

```diff
@@ -0,0 +1,262 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Downward APIs for resource limits and requests
+
+## Background
+
+Currently downward APIs (via environment variables and volume plugin) only support exposing a Pod's name, namespace, annotations, labels and its IP. This document explains the need and design to extend them to expose resources (e.g. cpu, memory, storage) limits and requests.
+
+## Motivation
+
+By exposing information about computing resource requests & limits into pods, image/application authors can be aware of the computing environment in which the pods run, and could modify container images accordingly.
+
+## Design
+
+This is mostly driven by the discussion in https://github.com/kubernetes/kubernetes/issues/9473.
+
+### API with full json path selectors
+
+Full json path selectors specify the complete path to the resources limits and requests relative to pod spec.
+
+#### Environment variables
+
+This table shows how selectors can be used for various requests and limits to be exposed as environment variables.
```

Comment:
How would this work if we support dynamically updating resource limits in the future?
cc @bgrant0607